docs/modules/ROOT/pages/upgrade_to_version_2.adoc

```diff
@@ -0,0 +1,166 @@
+= Upgrade to Version 2.x
+:doctype: book
+
+== Configuration File Update
+
+In version 2.x:
+
+ - `RSpec/InvalidPredicateMatcher` cop is removed
+ - `CustomIncludeMethods` configuration option for `RSpec/EmptyExampleGroup` is removed
+ - cop departments are nested for cops with a department that doesn't match the extension name (`Capybara`, `FactoryBot`, `Rails`)
+
+[discrete]
+=== Adjust the configuration of `RSpec/EmptyExampleGroup`
+
+[source,yaml]
+----
+# .rubocop.yml
+
+# Before
+RSpec/EmptyExampleGroup:
+  CustomIncludeMethods:
+  - include_tests
+
+# After
+AllCops:
+  RSpec:
+    Language:
+      Includes:
+        Example:
+        - include_tests
+----
+
+=== Add a top-level `RSpec` department
+
+RuboCop extensions had cops with clashing names and departments, e.g. both `rspec-rails` and `rubocop-rspec` had `Rails::HttpStatus` cops.
+To avoid issues, e.g. inability to disable just one of the cops, each extension now has its own uber-department.
+Expectedly, RuboCop RSpec's uber-department name is `RSpec`.
+Changes are only applied to cops that don't already have the department set to `RSpec`, i.e. `Capybara`, `FactoryBot` and `Rails`.
+
+[source,yaml]
+----
+# .rubocop.yml
+
+# Before
+Capybara/CurrentPathExpectation:
+  Enabled: false
+
+FactoryBot/AttributeDefinedStatically:
+  Enabled: false
+
+# remains the same
+RSpec/EmptyExampleGroup:
+  Enabled: false
+
+# After
+RSpec/Capybara/CurrentPathExpectation:
+  Enabled: false
+
+RSpec/FactoryBot/AttributeDefinedStatically:
+  Enabled: false
+
+# remains the same
+RSpec/EmptyExampleGroup:
+  Enabled: false
+----
+
+https://github.com/rubocop-hq/rubocop/pull/8490[Learn more about this change].
+
+== Custom Cop Update Guide
+
+Due to significant API changes custom cops will break.
+Here is the summary of the changes:
+
+1) The base class for cops is now `RuboCop::Cop::RSpec::Base` instead of `RuboCop::Cop::RSpec::Cop`.
```

Comment:
I think this line should start with `1.`
https://asciidoctor.org/docs/asciidoc-writers-guide/#ordering-the-things

Reply:
Parrotted it from here https://github.com/rubocop-hq/rubocop/blame/master/docs/modules/ROOT/pages/v1_upgrade_notes.adoc#L10 that is quite good on the eye https://docs.rubocop.org/rubocop/v1_upgrade_notes.html

Reply:
> quite good on the eye

Looks like I’ll have to open a pull request to fix RuboCop’s documentation too. 😄 

IMO this

<img width="403" alt="Screenshot 2020-10-22 at 22 27 09" src="https://user-images.githubusercontent.com/22333/96926409-cfa0f380-14b5-11eb-8df3-738c212a8e02.png">

looks more pleasing than this

<img width="403" alt="Screenshot 2020-10-22 at 22 22 31" src="https://user-images.githubusercontent.com/22333/96926424-d465a780-14b5-11eb-99f6-36a45b3f841e.png">

(note especially the indentation of the 2nd point)

Reply:
Again, sorry for my nit-pick comments.

Reply:
Agree, AsciiDoc's list looks better.

> Looks like I’ll have to open a pull request to fix RuboCop’s documentation too. 

:laughing: :shrug:

Reply:
> sorry for my nit-pick comments

No worries, those comments are all to the point. :+1: